Content.Shared/SS220/SharedTriggers/DamageOnTrigger/DamageOnTriggerSystem.cs

```diff
@@ -0,0 +1,30 @@
+// © SS220, An EULA/CLA with a hosting restriction, full text: https://raw.githubusercontent.com/SerbiaStrong-220/space-station-14/master/CLA.txt
+
+using Content.Shared.Damage;
+
+namespace Content.Shared.SS220.SharedTriggers.DamageOnTrigger;
+
+/// <summary>
+/// This handles deals damage when triggered
+/// </summary>
+public sealed class DamageOnTriggerSystem : EntitySystem
+{
+
+    [Dependency] private readonly DamageableSystem _damageableSystem = default!;
+    /// <inheritdoc/>
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<DamageOnTriggerComponent, SharedTriggerEvent.SharedTriggerEvent>(OnCollide);
+    }
+
+    private void OnCollide(Entity<DamageOnTriggerComponent> ent, ref SharedTriggerEvent.SharedTriggerEvent args)
```

Comment:
или просто OnTrigger
```suggestion
    private void OnTriggered(Entity<DamageOnTriggerComponent> ent, ref SharedTriggerEvent.SharedTriggerEvent args)
```